Add remote store buffer and index interval settings (#4973)
opensearch-project/documentation-website: _api-reference/cluster-api/cluster-settings.md

Parameter | Data type | Description
:--- | :--- | :---
`flat_settings` | Boolean | Whether to return settings in the flat form, which can improve readability, especially for heavily nested settings. For example, the flat form of `"cluster": { "max_shards_per_node": 500 }` is `"cluster.max_shards_per_node": "500"`.
`include_defaults (GET only)` | Boolean | Whether to include default settings as part of the response. This parameter is useful for identifying the names and current values of settings you want to update.
`cluster_manager_timeout` | Time unit | The amount of time to wait for a response from the cluster manager node. Default is `30 seconds`.
`timeout` (PUT only) | Time unit | The amount of time to wait for a response from the cluster. Default is `30 seconds`.

> 

Author: I removed the formatting changes.

| `cluster.persistent_tasks.allocation.enable` | String | Enables or disables allocation for persistent tasks: <br /> <br /> `all` – Allows persistent tasks to be assigned to nodes. <br /> <br /> `none` – No allocations are allowed for persistent tasks. This does not affect persistent tasks already running. <br /> <br /> Default is `all`. |
| `cluster.persistent_tasks.allocation.recheck_interval` | Time unit | The cluster manager automatically checks whether or not persistent tasks need to be assigned when the cluster state changes in a significant way. There are other factors, such as memory usage, that will affect whether or not persistent tasks are assigned to nodes but do not otherwise cause the cluster state to change. This setting defines how often assignment checks are performed in response to these factors. Default is `30 seconds`, with a minimum of `10 seconds` being required. |
| `cluster_minimum_index_refresh_interval` | Time unit | Sets the minimum index refresh interval. Default is `10s`.

> `cluster_default_index_refresh_interval` | Time unit | Sets the default index refresh interval in seconds.

Reviewer: The setting keys are incorrect, pls update with the correct names.

| network.breaker.inflight_requests.limit | String | The limit for the in-flight requests breaker. Default is `100%` of the JVM heap. |
| network.breaker.inflight_requests.overhead | Integer/Time unit | The constant that all in-flight request estimations are multiplied by to determine a final estimation. Default is `2`. |
| script.max_compilations_rate | String | The limit for the number of unique dynamic scripts within a defined interval that are allowed to be compiled. Default is 150 every 5 minutes: `150/5m`. |

> | cluster.default.index.refresh_interval | Time unit | Sets the refresh interval when the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You cannot set the interval lower than the `cluster.minimum.index.refresh_interval` setting. |

Reviewer 2: ```suggestion
| cluster.default.index.refresh_interval | Time unit | Sets the refresh interval when the `index.refresh_interval` setting is not provided. This setting can be useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You cannot set the interval lower than the `cluster.minimum.index.refresh_interval` setting. |
```

> | cluster.default.index.refresh_interval | Time unit | Sets the refresh interval what the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You can only set the interval as low as the `cluster.minimum.index.refresh_interval` setting. |

Reviewer: `what the` -> `when the`

> | cluster.default.index.refresh_interval | Time unit | Sets the refresh interval what the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You can only set the interval as low as the `cluster.minimum.index.refresh_interval` setting. |

Reviewer 3: The first sentence didn't make sense. Check me on the suggested change. I'm not sure if that's the idea.

I would also break up the two ideas "set a default refresh interval across all indexes in a cluster" and "also support the `searchIdle` setting" into two separate sentences. It can do two things. But, since they seem not at all connected, having them in the same sentence reads awkwardly. Can you also link to a description of `searchidle`? It' not clear how the setting supports it.
```suggestion
| cluster.default.index.refresh_interval | Time unit | Sets the refresh interval when the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You cannot set the interval lower than the `cluster.minimum.index.refresh_interval` setting. |
```

> | cluster.default.index.refresh_interval | Time unit | Sets the refresh interval what the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You can only set the interval as low as the `cluster.minimum.index.refresh_interval` setting. |

Author: ```suggestion
| cluster.default.index.refresh_interval | Time unit | Sets the refresh interval when the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You can only set the interval as low as the `cluster.minimum.index.refresh_interval` setting. |
```

> | cluster.default.index.refresh_interval | Time unit | Sets the refresh interval when the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You can only set the interval as low as the `cluster.minimum.index.refresh_interval` setting. |

Reviewer 3: Is it possible to link to the `searchIdle` setting? How does this setting support it?

```suggestion
| cluster.default.index.refresh_interval | Time unit | Sets the refresh interval when the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You cannot set the interval lower than the `cluster.minimum.index.refresh_interval` setting. |
```

> | cluster.default.index.refresh_interval | Time unit | Sets the refresh interval what the `index.refresh_interval` setting is not provided when the index was created of what the existing `index.refresh_interval` setting is set to `null`. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You can only set the interval as low as the `cluster.minimum.index.refresh_interval` setting. |

Reviewer: `Sets the refresh interval what the `index.refresh_interval` setting is not provided when the index was created of what the existing `index.refresh_interval` setting is set to `null`.` - Finding it difficult to make sense of this.

> | cluster.default.index.refresh_interval | Time unit | Sets the refresh interval what the `index.refresh_interval` setting is not provided when the index was created of what the existing `index.refresh_interval` setting is set to `null`. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You can only set the interval as low as the `cluster.minimum.index.refresh_interval` setting. |

Author: ```suggestion
| cluster.default.index.refresh_interval | Time unit | Sets the refresh interval what the `index.refresh_interval` setting is not provided. This setting is useful when you want to set a default refresh interval across all indexes in a cluster and also support the `searchIdle` setting. You can only set the interval as low as the `cluster.minimum.index.refresh_interval` setting. |
```

> | cluster.minimum.index.refresh_interval | Time unit | Sets the minimum refresh interval applicable for all indexes in the cluster. The `cluster.default.index.refresh_interval` setting should be higher than this setting's value. If, during index creation, the `index.refresh_interval` setting is lower than the minimum set, index creation fails. |

Reviewer 3: ```suggestion
| cluster.minimum.index.refresh_interval | Time unit | Sets the minimum refresh interval and applies it to all indexes in the cluster. The `cluster.default.index.refresh_interval` setting should be higher than this setting's value. If, during index creation, the `index.refresh_interval` setting is lower than the minimum set, index creation fails. |
```

> | cluster.remote_store.translog.buffer_interval | Time unit | The default value of the translog buffer interval used when performing periodic translog updates. Use this setting only when the `index.remote_store.translog.buffer_interval` is not set inside the index's settings |

Reviewer: `The default value of the translog buffer interval used when performing periodic translog updates.` -> The last section can be updated to `periodic updates to remote store.`

`Use this setting only when the `index.remote_store.translog.buffer_interval` is not set inside the index's settings` - This conveys a different meaning. Please refer https://github.com/opensearch-project/documentation-website/pull/4973#issuecomment-1711233429. `The setting is only effective if the index setting index.remote_store.translog.buffer_interval is not present with an index.`
